src/charts/tooltip/TooltipComponent.js

```diff
@@ -89,7 +88,7 @@ export default class TooltipComponent extends React.Component {
             y,
         }));
 
-        let {children: {props: {customMouseMove}}} = this.props;
+        let {customMouseMove} = this.props;
```

Comment:
This refactor required this change. Now the `customMouse` props will need to be given to `TooltipComponent` instead of `StackedAreaComponent`.

This is because we can't access the props of the `childChart` anymore since we are overwriting them in the constructor.

I thought about trying to pull out the props from `childChart` first, before overwriting them, but I couldn't figure out a way to do it without having to instantiate `childChart` twice.

Not sure if this is an improvement over the old implementation, but the code does look a bit cleaner.

Reply:
It looks OK, let's go with it for now.